### PR TITLE
Added retry fetchdata,  meaningful error messages and query filter to…

### PR DESCRIPTION
**Pull request by maayarosama**
#Description

Added retrying to fetchData,  meaningful error messages, query filter to fetch only online results and refresh button.

#Related issues

https://github.com/threefoldtech/tfgrid_stats/issues/18

### CODE REVIEW

File: src/App.svelte [hunk 1]
```diff
@@ -5,29 +5,49 @@
 
   let data: IStatsRes;
   let loading = true;
+  let error: string;
 
-  onMount(async () => {
+  $: refresh = async () => {
```
Comment (MohamedElmdary) [inline src/App.svelte]: Why did you use reactive statement here ?
I think it should be turned into regular variable assignment

File: src/App.svelte [hunk 2]
```diff
@@ -5,29 +5,49 @@
...
+      console.log("Statistics couldn't load due to:", err);
     } finally {
-      try {
-        data = await fetchData();
-      } catch (err) {
-        console.log("Error", err);
-      } finally {
-        loading = false;
-      }
+      loading = false;
```
Comment (MohamedElmdary) [inline src/App.svelte:21]: Please make sure to fix loading at the initial start of the page
Make sure its on the center of the screen

File: src/App.svelte [hunk 3]
```diff
@@ -5,29 +5,49 @@
...
 </script>
 
 <main>
   {#if loading}
     <div class:lds-dual-ring={loading} />
   {:else if data}
-    <h2 class="node-title">Node Distribution</h2>
+    <div>
+      <button class="refresh" on:click={refresh}>
+        <svg
```
Comment (MohamedElmdary) [inline src/App.svelte]: Please add a rotate animation and disable the refresh button whenever the user click refresh

File: src/App.svelte [hunk 4]
```diff
@@ -5,29 +5,49 @@
...
+          viewBox="0 0 16 16"
+        >
+          <path
+            fill-rule="evenodd"
+            d="M8 3a5 5 0 1 0 4.546 2.914.5.5 0 0 1 .908-.417A6 6 0 1 1 8 2v1z"
+          />
+          <path
+            d="M8 4.466V.534a.25.25 0 0 1 .41-.192l2.36 1.966c.12.1.12.284 0 .384L8.41 4.658A.25.25 0 0 1 8 4.466z"
+          />
+        </svg></button
```
Comment (MohamedElmdary) [inline src/App.svelte]: Check if data is returned or if its ignored due to failure to catch it
should be applied to dev, test, main and qa

File: src/App.svelte (hunk 2 above)
Comment (maayarosama) [reply src/App.svelte:17]: @MohamedElmdary This's mentioned in another issue [](https://github.com/threefoldtech/tfgrid_stats/issues/17).

Comment (maayarosama) [reply src/App.svelte:21]: @MohamedElmdary This's mentioned in this issue #17, and should be resolved soon.

**Comment by maayarosama** [comment]
The styling of the loading icon mentioned in this issue #17 is solved here.

**Pull request merged**